### PR TITLE
Disable export for unsupported units

### PR DESCRIPTION
**Pull request by haticekaratay**
### Description

This pull request disables writing data products to FITS files due to unsupported DN/s units in Astroquery's internal.
image

Fixes #<Issue Number>

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Review by kecnry — APPROVED** [review]
Just one small comment, otherwise looks good to me once tests pass!

File: jdaviz/configs/default/plugins/export/tests/test_export.py [hunk 1]
```diff
@@ -204,6 +204,28 @@ def test_export_data(cubeviz_helper, spectrum1d_cube):
     assert ep.data_invalid_msg == ''
 
 
+def test_disable_export_for_unsupported_units(specviz2d_helper):
+    from astropy.units import def_unit, Unit
+    dn = def_unit("DN", represents=Unit("count"))
```
Comment (cshanahan1) [inline jdaviz/configs/default/plugins/export/tests/test_export.py:209]: I think you can just use the existing u.DN instead of defining a new one

**Review by cshanahan1 — APPROVED** [review]
approved, pending small comments

File: jdaviz/configs/default/plugins/export/export.py [hunk 2]
```diff
@@ -252,14 +252,13 @@ def _set_subset_not_supported_msg(self, msg=None):
...
         if self.dataset.selected_obj is not None:
-            data_unit = self.dataset.selected_obj.unit
             if self.dataset.selected_obj.meta.get("Plugin", None) is None:
                 # NOTE: should not be a valid choice due to dataset filters, but we'll include
                 # another check here.
                 self.data_invalid_msg = "Data export is only available for plugin generated data."
             elif not isinstance(self.dataset.selected_obj, (Spectrum1D, CCDData)):
                 self.data_invalid_msg = "Export is not implemented for this type of data"
-            elif data_unit == u.Unit('DN/s'):
+            elif (data_unit := self.dataset.selected_obj.unit) == u.Unit('DN/s'):
```
Comment (kecnry) [inline jdaviz/configs/default/plugins/export/export.py:261]: WALRUS!!! 🎉 😁

Comment (cshanahan1) [reply jdaviz/configs/default/plugins/export/export.py:261]: what is a walrus?!

Comment (cshanahan1) [reply jdaviz/configs/default/plugins/export/export.py:261]: ohhhh := := :=

**Pull request merged**